### PR TITLE
[14.0][MIG] product_sequence

### PR DESCRIPTION
**Pull request by watthanun**

### CODE REVIEW

**Review by benbrich — APPROVED** [review]
I have just checked the functionality on runbot. Erverything works like I know it. Thanks and thumbs up!

File: product_sequence/models/product_category.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            "name": "Product " + prefix,
+            "code": "product.product - " + prefix,
+            "padding": 5,
+            "prefix": prefix,
+            "company_id": False,
+        }
+        return vals
+
+    def write(self, vals):
+        prefix = vals.get("code_prefix")
```
Comment (ps-tubtim) [inline product_sequence/models/product_category.py]:
```suggestion
        prefix = vals.get("code_prefix", False)
```
I think you should add a default for `get` function.

File: product_sequence/models/product_category.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                if rec.sequence_id:
+                    rec.sudo().sequence_id.prefix = prefix
+                else:
+                    seq_vals = self._prepare_ir_sequence(prefix)
+                    rec.sequence_id = self.env["ir.sequence"].create(seq_vals)
+        return super().write(vals)
+
+    @api.model
+    def create(self, vals):
+        prefix = vals.get("code_prefix")
```
Comment (ps-tubtim) [inline product_sequence/models/product_category.py]:
```suggestion
        prefix = vals.get("code_prefix", False)
```
Like the above comment.

File: product_sequence/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        tracking=True,
+        help="Set to '/' and save if you want a new internal reference "
+        "to be proposed.",
+    )
+
+    @api.model
+    def create(self, vals):
+        if "default_code" not in vals or vals["default_code"] == "/":
+            categ_id = vals.get("categ_id")
+            template_id = vals.get("product_tmpl_id")
```
Comment (ps-tubtim) [inline product_sequence/models/product_product.py]:
```suggestion
            categ_id = vals.get("categ_id", False)
            template_id = vals.get("product_tmpl_id", False)
```
Like the above comment.

File: product_sequence/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2004 Tiny SPRL
+# Copyright 2016 Sodexis
+# Copyright 2018 ForgeFlow S.L.
+#   (http://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Product Sequence",
+    "version": "14.0.1.0.0",
+    "author": "Zikzakmedia SL,Sodexis,Odoo Community Association (OCA)",
```
Comment (ps-tubtim) [inline product_sequence/__manifest__.py]:
```suggestion
    "author": "Zikzakmedia SL, Sodexis, Odoo Community Association (OCA)",
```

**Review by ps-tubtim — COMMENTED** [review]
I have a few comments.

**Review by ps-tubtim — APPROVED** [review]

**Pull request merged**